source/opt/set_spec_constant_default_value_pass.cpp

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "set_spec_constant_default_value_pass.h"
+
+#include <cstring>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+#include "util/parse_number.h"
+
+#include "def_use_manager.h"
+#include "spirv-tools/libspirv.h"
+#include "type_manager.h"
+#include "types.h"
+
+namespace spvtools {
+namespace opt {
+
+namespace {
+using spvutils::NumberType;
+using spvutils::EncodeNumberStatus;
+using spvutils::ParseAndEncodeNumber;
+
+// Given a numeric value in null-terminated c string form and the expected type
+// of the value, parse the string and encode it in a vector of words. If the
+// value is a scalar integer or floating point value, encodes the value in
+// SPIR-V encoding format. If the value is 'false' or 'true', returns a vector
+// with single word with value 0 or 1 respectively. Returns the vector
+// containing the encoded value on success. Otherwise returns an empty vector.
+std::vector<uint32_t> ParseDefaultValueStr(const char* text,
+                                           const analysis::Type* type) {
+  std::vector<uint32_t> result;
+  if (!strcmp(text, "true")) {
+    result.push_back(1u);
+  } else if (!strcmp(text, "false")) {
+    result.push_back(0u);
+  } else {
+    NumberType number_type = {32, SPV_NUMBER_SIGNED_INT};
+    if (const auto* IT = type->AsInteger()) {
+      number_type.bitwidth = IT->width();
+      number_type.kind =
```

Comment:
Actually, I *DO* like the code the way it is.   That's because it's less sensitive to a change of the initializer.  (And I *do* want the initializer changed.)